src/Raven.Server/Documents/Patch/ScriptRunner.cs

```diff
@@ -230,6 +237,174 @@ public SingleRun(DocumentDatabase database, RavenConfiguration configuration, Sc
                 }
             }
 
+            private (string Id, BlittableJsonReaderObject Doc) GetIdAndDocFromArg(JsValue firstArgs, string signature)
+            {
+                if (firstArgs.IsObject() && firstArgs.AsObject() is BlittableObjectInstance doc)
+                    return (doc.DocumentId, doc.Blittable);
+
+                if (firstArgs.IsString())
+                {
+                    var id = firstArgs.AsString();
+                    var document = _database.DocumentsStorage.Get(_docsCtx, id);
+                    if (document == null)
+                        throw new DocumentDoesNotExistException(id, "Cannot operate on a missing document.");
+
+                    return (id, document.Data);
+                }
+
+                throw new InvalidOperationException($"{signature}: 'doc' must be a string argument (the document id) or the actual document instance itself");
+            }
+            
+            private string GetIdFromArg(JsValue firstArgs, string signature)
+            {
+                if (firstArgs.IsObject() && firstArgs.AsObject() is BlittableObjectInstance doc)
+                    return doc.DocumentId;
+
+                if (firstArgs.IsString())
+                {
+                    var id = firstArgs.AsString();
+                    return id;
+                }
+
+                throw new InvalidOperationException($"{signature}: 'doc' must be a string argument (the document id) or the actual document instance itself");
+            }
+            
+            private static string GetStringArg(JsValue jsArg, string signature, string argName)
+            {
+                if (jsArg.IsString() == false)
+                    throw new ArgumentException($"{signature}: The {argName} argument should be a string, but got {jsArg.GetType().Name}");
```

Comment:
You are getting the .NET type of the `JsValue`, which will be `JsValue`, you need to get the Jint type here.

Reply:
It is because I found that both the `objectInstance` and `arrayInstance` return Object type.
So Just not lose the information I will return Jint type and .NET type, OK?

Reply:
ok